pkg/gitops/update_files.go

```diff
@@ -0,0 +1,76 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"log"
+)
+
+// UpdateFilesParams are parameters for UpdateFiles function.
+type UpdateFilesParams struct {
```

Comment:
Wdyt about separating the dependencies from the actual params? That way it would be clearer to see what needs to be mocked in tests

Reply:
I'm not against doing that, but I wouldn't do that for the tests only, I would rather do this to have a better/cleaner API for the package.

For the tests only part, my view is: nothing has to be mocked, the test can actually bootstrap every concrete implementation (no mocks at all). It's the test's responsibility to determine what it's scope is: does it only want to test the `UpdateFiles` function (like `update_files_test.go`)/does it want to test everything except calling Github API/does it want to call Github API as well for a true e2e flow/etc.

What do you think, and what is your concrete suggestion for `UpdateFiles` API if you still think that separating is a cleaner approach?

Reply:
Hmm I'm not sure about this, I'm against having 3+ positional parameters for a function, named parameters are cleaner for my taste. I'm happy to talk about this next week in person as well/compare different approaches side-by-side to get the cleanest result eventually :)